build.sbt

```diff
@@ -8,21 +8,25 @@ lazy val commonSettings = Seq(
     case PathList("META-INF", "MANIFEST.MF") => MergeStrategy.discard
     case _ => MergeStrategy.first}},
   scalacOptions ++= Seq("-deprecation","-unchecked","-Xsource:2.11"),
-  libraryDependencies += "org.scalatest" %% "scalatest" % "3.0.1" % "test",
-  libraryDependencies += "org.json4s" %% "json4s-native" % "3.5.3",
+  libraryDependencies += "org.scalatest" %% "scalatest" % "3.0.5" % "test",
+  libraryDependencies += "org.json4s" %% "json4s-native" % "3.6.1",
   libraryDependencies += "org.scala-lang" % "scala-reflect" % scalaVersion.value,
   libraryDependencies += "edu.berkeley.cs" %% "firrtl-interpreter" % "1.2-SNAPSHOT",
-  libraryDependencies += "com.github.scopt" %% "scopt" % "3.7.1",
+  libraryDependencies += "com.github.scopt" %% "scopt" % "3.7.0",
   addCompilerPlugin("org.scalamacros" % "paradise" % "2.1.0" cross CrossVersion.full),
   resolvers ++= Seq(
     Resolver.sonatypeRepo("snapshots"),
     Resolver.sonatypeRepo("releases"),
     Resolver.mavenLocal))
 
+lazy val rebarFirrtl = (project in file("tools/firrtl"))
```

Comment:
I'm not sure this does what you expect.  To the best of my knowledge, we need to manage the firrtl dependency manually, and this might just work because you have a stale firrtl.jar living in Rocket Chip's lib/ . Did you have a look at my PR / you need to change the Makefile.

Reply:
If you look at `common.mk` I deal with the `.jar` file in by creating it using the `tools/firrtl` version of `firrtl`. This is so that `hardfloat` can work correctly. Additionally, this new `firrtl` jar is put in `rocket-chip` and overrides `rocket-chip`s jar. This currently works for the builds. The only conflict that I can see is if someone runs `make` from within the `rocket-chip` dir in which case this `firrtl.jar` will get overriden.

This project is used mainly for the `barstools` projects since they depend on `firrtl`.

Reply:
The problem is that you need to put the jar at rebar/lib, not just rocket-chip/lib (which you might have to keep since it's currently a root project -- not sure). That will satisfy all subproject dependencies that require firrtl. 

In any case, if things build alright you can ship this, though i'd encourage you to do a fresh clone to check. The SBT stuff needs to get reworked for the firesim merge anyways.

Reply:
Yes. Fresh clone works with this. I double-checked. The only downside to this is that it will rebuild `firrtl` twice, I think (once for the initial jar build, and again for the project dependency in the main sbt build)

Reply:
Addressed offline: Currently `firrtl` is managed and unmanaged (building the jar in `make`). In an ideal world, it would be solely managed by `sbt` but for now we should just be unmanaged. This work (of choosing managed v unmanaged) will be done in the future `firesim` merge.